runtime/wasm/effect.wat

```diff
@@ -303,7 +306,13 @@
          (ref.func $do_perform)
          (struct.new $pair (local.get $eff) (local.get $cont))))
 
+   (global $effect_allowed (mut i32) (i32.const 1))
+
    (func (export "%perform") (param $eff (ref eq)) (result (ref eq))
+      (if (i32.eqz (global.get $effect_allowed))
```

Comment:
I dont see any equivalent logic on the js side. I seems that caml_assume_no_perform completely forbid effect with wasm while effects could still happen with js after resuming a stack ? I must admit my understanding is poor in the area.

Reply:
In both JS and Wasm, performing under `caml_assume_no_perform` must raise `Effect.Unhandled`, expect after installing a handler, i.e. resuming a stack. On the js_of_ocaml side, it is ensured by having caml_assume_no_perform be a synonym of `caml_callback`, which switches to a blank fiber stack. On the wasm_of_ocaml side, it’s not possible to do it for a reason that @vouillon explained me but that I forgot. An alternative is to have this global boolean to track whether performing an effect should raise.

Reply:
So we have different behavior between wasm and js ?

Reply:
No, the behavior is the same, as checked by the tests in `tests-ocaml/lib-effects/`.